utils.go

```diff
@@ -556,6 +577,39 @@ func longestCommonPath(paths []string) string {
 	return shortest
 }
 
+// Merges 'src' path into 'dst' one attending to their largest-common-paths.
```

Comment:
Don't understand the need for this ... call me to talk please.

Reply:
Yep, it's not trivial. Ttyl.